FWCore/Framework/interface/TriggerResultsBasedEventSelector.h

```diff
@@ -18,91 +18,50 @@
 #include "DataFormats/Common/interface/Handle.h"
 #include "FWCore/Utilities/interface/InputTag.h"
 
-namespace edm
-{
+namespace edm {
   class ModuleCallingContext;
 
-  namespace detail
-  {
+  namespace detail {
     typedef edm::Handle<edm::TriggerResults> handle_t;
 
-    class NamedEventSelector
-    {
+    class NamedEventSelector {
     public:
       NamedEventSelector(std::string const& n, EventSelector const& s) :
-	inputTag_("TriggerResults", "", n), 
-	eventSelector_(s), 
-	product_() 
+	inputTag_("TriggerResults", "", n),
+	eventSelector_(s)
       { }
 
-      void fill(EventPrincipal const& e, ModuleCallingContext const* mcc);
-
-      bool match()
-      {
-	return eventSelector_.acceptEvent(*product_);
+      bool match(TriggerResults const& product) {
```

Comment:
Not yet. Work is needed in class "EventSelector" before "match", and a whole bunch of other functions can be made const. That is the goal, but we are not there yet.